program/rust/src/tests/test_upd_price.rs

```diff
@@ -0,0 +1,292 @@
+            clock_account.clone()
+        ],
+        &instruction_data
+    )
+    .is_ok());
+
+    {
+        let price_data = load_checked::<pc_price_t>(&price_account, PC_VERSION).unwrap();
+        assert_eq!(price_data.comp_[0].latest_.price_, 81);
+        assert_eq!(price_data.comp_[0].latest_.conf_, 2);
+        assert_eq!(price_data.comp_[0].latest_.pub_slot_, 3);
+        assert_eq!(price_data.comp_[0].latest_.status_, PC_STATUS_TRADING);
+        assert_eq!(price_data.valid_slot_, 3);
+        assert_eq!(price_data.agg_.pub_slot_, 4);
+        assert_eq!(price_data.agg_.price_, 81);
+        assert_eq!(price_data.agg_.status_, PC_STATUS_TRADING);
+    }
+
+    // next price doesnt change and neither does aggregate but slot does
+    populate_instruction(&mut instruction_data, None, None, Some(4));
+    update_clock_slot(&mut clock_account, 5);
+    assert!(upd_price(
+        &program_id,
+        &[
+            funding_account.clone(),
+            price_account.clone(),
+            clock_account.clone()
+        ],
+        &instruction_data
+    )
+    .is_ok());
+
+    {
+        let price_data = load_checked::<pc_price_t>(&price_account, PC_VERSION).unwrap();
+        assert_eq!(price_data.comp_[0].latest_.price_, 81);
+        assert_eq!(price_data.comp_[0].latest_.conf_, 2);
+        assert_eq!(price_data.comp_[0].latest_.pub_slot_, 4);
+        assert_eq!(price_data.comp_[0].latest_.status_, PC_STATUS_TRADING);
+        assert_eq!(price_data.valid_slot_, 4);
+        assert_eq!(price_data.agg_.pub_slot_, 5);
+        assert_eq!(price_data.agg_.price_, 81);
+        assert_eq!(price_data.agg_.status_, PC_STATUS_TRADING);
+    }
+
+    // try to publish back-in-time
+    populate_instruction(&mut instruction_data, None, None, Some(1));
+    update_clock_slot(&mut clock_account, 5);
+    assert_eq!(
+        upd_price(
+            &program_id,
+            &[
+                funding_account.clone(),
+                price_account.clone(),
+                clock_account.clone()
+            ],
+            &instruction_data
+        ),
+        Err(ProgramError::InvalidArgument)
+    );
+
+    {
+        let price_data = load_checked::<pc_price_t>(&price_account, PC_VERSION).unwrap();
+        assert_eq!(price_data.comp_[0].latest_.price_, 81);
+        assert_eq!(price_data.comp_[0].latest_.conf_, 2);
+        assert_eq!(price_data.comp_[0].latest_.pub_slot_, 4);
+        assert_eq!(price_data.comp_[0].latest_.status_, PC_STATUS_TRADING);
+        assert_eq!(price_data.valid_slot_, 4);
+        assert_eq!(price_data.agg_.pub_slot_, 5);
+        assert_eq!(price_data.agg_.price_, 81);
+        assert_eq!(price_data.agg_.status_, PC_STATUS_TRADING);
+    }
+
+    populate_instruction(&mut instruction_data, Some(50), Some(6), Some(5));
+    update_clock_slot(&mut clock_account, 6);
+
+    // Publishing a wide CI results in a status of unknown.
+
+    // check that someone doesn't accidentally break the test.
```

Comment:
I think you dropped an assert for this comment -- the point is to validate that the publisher's status is TRADING before running the instruction that sets their status to UNKNOWN. If you don't have this assertion, it's possible that a change to something before this test makes the test meaningless.

Reply:
Readded the assert